cfme/automate/service_dialogs.py

```diff
@@ -77,166 +22,207 @@
-            if not dynamic_chkbox:
-                entry_table.click_cell(header='value', value='<New Entry>')
-                fill(element_form, {'entry_value': "Yes",
-                                    'entry_description': "entry_desc"},
-                    action=element_form.add_entry_button)
-            else:
-                node1 = "InspectMe"
-                sel.click(element_form.field_entry_point)
-                if self.appliance.version < "5.7":
-                    dynamic_tree.click_path("Datastore", "new_domain", "System", "Request", node1)
-                else:
-                    bt_tree.click_path("Datastore", "new_domain", "System", "Request", node1)
-                sel.click(element_form.apply_btn)
-                fill(element_form, {'field_show_refresh_button': True})
-        if choose_type == "Text Area Box":
-            text_area_table.click_cell(1, value="Default text")
-
-    def element(self, element_data):
-        return sel.element('//div[@class="panel-heading"]'
-            '[contains(normalize-space(.), "{}")]/..'.format(element_data))
-
-    def reorder_elements(self, tab, box, *element_data):
-        navigate_to(self, 'Edit')
-        tree = accordion.tree("Dialog")
-        tree.click_path(self.label, tab, box)
-        list_ele = []
-        for each_element in element_data:
-            list_ele.append(each_element.get("ele_label"))
-        ele_1 = self.element(list_ele[0])
-        ele_2 = self.element(list_ele[1])
-        sel.drag_and_drop(ele_1, ele_2)
-        sel.click(form_buttons.save)
-        flash.assert_no_errors()
-
-
-@navigator.register(ServiceDialog, 'All')
-class ServiceDialogAll(CFMENavigateStep):
+            view.flash.assert_message(
+                'Edit of Dialog "{}" was cancelled by the user'.format(self.label))
+
+    def delete(self):
+        """ Delete dialog method"""
+        view = navigate_to(self, "Details")
+        view.configuration.item_select('Remove Dialog', handle_alert=True)
+        view = self.create_view(DialogsView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        view.flash.assert_success_message(
+            'Dialog "{}": Delete successful'.format(self.label))
+
+    @property
+    def exists(self):
+        """ Returns True if dialog exists"""
+        try:
+            navigate_to(self, 'Details')
+            return True
+        except (CandidateNotFound, ItemNotFound):
+            return False
+
+    def delete_if_exists(self):
+        if self.exists:
+            self.delete()
+
+
+@navigator.register(DialogCollection)
+class All(CFMENavigateStep):
+    VIEW = DialogsView
+
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
         if self.obj.appliance.version < '5.8':
             self.prerequisite_view.navigation.select('Automate', 'Customization')
+            self.view.service_dialogs.tree.click_path(*self.obj.tree_path)
```

Comment:
You can replace the prerequisite in the step with
```
prerequisite = NavigateToAttribute('appliance.server', 'AutomateCustomization')
```

And thenyou can remove the navigation.select references, effectively removing the if completely.